File: src/main/java/gr/apt/persistence/entity/notification/NotificationRecipientRoles.java
package gr.apt.persistence.entity.notification;

import gr.apt.persistence.entity.superclass.AbstractEntity;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigInteger;
import java.sql.Timestamp;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "notification_recipient_roles", schema = "lms")
public class NotificationRecipientRoles extends AbstractEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE,generator = "notification_recipient_roles_seq")
    @SequenceGenerator(name = "notification_recipient_roles_seq",sequenceName = "lms.lms_notification_recipient_roles_id_seq")
    @Column(name = "id", nullable = false, precision = 0)
    private BigInteger id;
    @Basic
    @Column(name = "role_id", nullable = false, precision = 0)
    private BigInteger roleId;
    @Basic
    @Column(name = "notif_id", nullable = false, precision = 0)
    private BigInteger notifId;

    @ManyToOne
    @JoinColumn(name = "notif_id", referencedColumnName = "id", insertable = false, updatable = false)
    private Notification notificationByNotifId;

    public NotificationRecipientRoles(BigInteger roleId, BigInteger notifId) {
        this.roleId = roleId;
        this.notifId = notifId;
    }
}

File: src/main/java/gr/apt/lms/persistence/entity/notification/NotificationRecipientRoles.kt
package gr.apt.lms.persistence.entity.notification

import gr.apt.lms.persistence.entity.superclass.AbstractEntity
import java.math.BigInteger
import javax.persistence.*

@Entity
@Table(name = "notification_recipient_roles", schema = "lms")
open class NotificationRecipientRoles(
    @field:Column(
        name = "role_id",
        nullable = false,
        precision = 0
    ) @field:Basic open var roleId: BigInteger?, @field:Column(
        name = "notif_id",
        nullable = false,
        precision = 0
    ) @field:Basic open var notifId: BigInteger?
) : AbstractEntity() {
    constructor() : this(null, null)

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "notification_recipient_roles_seq")
    @SequenceGenerator(
        name = "notification_recipient_roles_seq",
        sequenceName = "lms.lms_notification_recipient_roles_id_seq",
        allocationSize = 1
    )
    @Column(name = "id", nullable = false, precision = 0)
    open var id: BigInteger? = null

//    @ManyToOne
//    @JoinColumn(name = "notif_id", referencedColumnName = "id", insertable = false, updatable = false)
//    open var notificationByNotifId: Notification? = null
}